src/State.ts

```diff
@@ -171,13 +169,12 @@ export class State<
       if (!stateValue.actions.length) {
         return stateValue as State<TC, TE>;
       }
-      const event = { type: ActionTypes.Init } as TE;
+      const _event = initEvent as SCXML.Event<TE>;
 
       return new State<TC, TE>({
         value: stateValue.value,
         context,
-        event,
```

Comment:
this is duplicated info, as `_event` always contain `event` in its `_event.data`, no need pass both here - both are assigned in State constructor though (backwar compat)